common/datablocks/src/kernels/data_block_scatter.rs

```diff
@@ -2,53 +2,41 @@
 //
 // SPDX-License-Identifier: Apache-2.0.
 
-use common_datavalues::DataArrayScatter;
-use common_datavalues::DataColumnarValue;
-use common_exception::ErrorCode;
+use common_datavalues::prelude::*;
 use common_exception::Result;
 
 use crate::DataBlock;
 
 impl DataBlock {
     pub fn scatter_block(
         block: &DataBlock,
-        indices: &DataColumnarValue,
+        indices: &DataColumn,
         scatter_size: usize,
     ) -> Result<Vec<DataBlock>> {
-        let columns_size = block.num_columns();
-        let mut scattered_columns: Vec<Option<DataColumnarValue>> = vec![];
+        let array = indices.to_array()?;
+        let array = array.u64()?;
 
-        scattered_columns.resize_with(scatter_size * columns_size, || None);
+        let columns_size = block.num_columns();
+        let mut scattered_columns = Vec::with_capacity(scatter_size);
 
         for column_index in 0..columns_size {
-            let column = block.column(column_index);
-            let scattered_column = DataArrayScatter::scatter(&column, &indices, scatter_size)?;
-
-            for scattered_index in 0..scattered_column.len() {
-                scattered_columns[scattered_index * columns_size + column_index] =
-                    Some(scattered_column[scattered_index].clone());
-            }
+            let mut indices = array.into_no_null_iter();
+
+            let columns = unsafe {
```

Comment:
Why need unsafe?

Reply:
scatter_unchecked is unsafe function, we don't need to check the index bound